src/contentBlock.ts

```diff
@@ -0,0 +1,50 @@
+import { z } from "zod";
+
+/**
+ * Content block schema for standard content blocks.
+ */
+export function contentBlock(): ReturnType<
+  typeof makeContentBlockQuery<typeof baseMarkdefsType>
+>;
+export function contentBlock<T extends z.ZodType>(args: {
+  markDefs: T;
+}): ReturnType<typeof makeContentBlockQuery<T>>;
+export function contentBlock({ markDefs }: { markDefs?: z.ZodType } = {}) {
+  return makeContentBlockQuery(markDefs || baseMarkdefsType);
+}
+
+export function contentBlocks(): z.ZodArray<
+  ReturnType<typeof makeContentBlockQuery<typeof baseMarkdefsType>>
+>;
+export function contentBlocks<T extends z.ZodType>(args: {
+  markDefs: T;
+}): z.ZodArray<ReturnType<typeof makeContentBlockQuery<T>>>;
+export function contentBlocks({ markDefs }: { markDefs?: z.ZodType } = {}) {
+  return z.array(makeContentBlockQuery(markDefs || baseMarkdefsType));
+}
+
+function makeContentBlockQuery<T extends z.ZodType>(markDefs: T) {
```

Comment:
Do we need to prevent  extending `T` from `z.ZodType` allowing collection types? It's my understanding these would not be nested objects.